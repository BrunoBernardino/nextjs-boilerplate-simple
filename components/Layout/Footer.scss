@import '__variables';

.Footer {
  display: block;
  padding: 2rem 0;
  margin: 0 auto;
  text-align: center;

  &__nav__items {
    display: block;

    @media #{$bigger-screen} {
      @include flex;
    }
  }

  &__nav__item {
    margin: 0 auto 2rem;

    @media #{$bigger-screen} {
      margin: 0 0 1rem;

      &:after {
        content: '/';
        display: inline-block;
        margin: 0 2rem;
        font-weight: 300;
      }
    }

    &:last-child {
      &:after {
        display: none;
      }
    }

    a {
      text-decoration: none;
      font-weight: bold;
    }
  }

  &__title {
    font-size: 0.8rem;
    line-height: 1rem;
    font-weight: 300;
    margin-top: 3rem;

    @media #{$bigger-screen} {
      margin-top: 1rem;
    }

    a {
      text-decoration: none;
    }
  }
}
