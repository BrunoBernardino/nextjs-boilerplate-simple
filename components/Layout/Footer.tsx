import React from 'react';

import { githubUrl } from 'lib/constants';

import './Footer.scss';

const Footer = () => {
  return (
    <footer className="Footer">
      <nav className="Footer__nav">
        <ul className="Footer__nav__items">
          <li className="Footer__nav__item">
            view source in <a href={githubUrl}>GitHub</a>
          </li>
        </ul>
      </nav>

      <h3 className="Footer__title">
        by <a href="https://brunobernardino.com">Bruno Bernardino</a>
      </h3>
    </footer>
  );
};

export default Footer;
