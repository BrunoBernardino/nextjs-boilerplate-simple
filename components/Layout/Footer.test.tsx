import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import expect from 'expect';
import enzymify from 'expect-enzyme';

import Footer from './Footer';

expect.extend(enzymify());

describe('Footer', () => {
  it('renders the Footer without errors', () => {
    const wrapper = shallow(<Footer />);
    expect(wrapper.find('Link').length).toEqual(0);
    expect(
      wrapper.find('a[href^="https://github.com/BrunoBernardino"]').length,
    ).toEqual(1);
  });

  it('renders the footer as expected', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
