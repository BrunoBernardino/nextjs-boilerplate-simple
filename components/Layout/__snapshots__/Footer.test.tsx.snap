// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the footer as expected 1`] = `
<footer
  className="Footer"
>
  <nav
    className="Footer__nav"
  >
    <ul
      className="Footer__nav__items"
    >
      <li
        className="Footer__nav__item"
      >
        view source in 
        <a
          href="https://github.com/BrunoBernardino/nextjs-boilerplate-simple"
        >
          GitHub
        </a>
      </li>
    </ul>
  </nav>
  <h3
    className="Footer__title"
  >
    by 
    <a
      href="https://brunobernardino.com"
    >
      Bruno Bernardino
    </a>
  </h3>
</footer>
`;
